### PR TITLE
Sharing key increase complexity

### PR DESCRIPTION
**Pull request by VirtualGit**
Add a sharing key complexity setting to generate more complex keys.
Keys too long will be displayed with an ellipsis.
(refs #883)
That's my first github PR. I updated my branch twice so there are merge commits. Tell me if this is ok or if you want I make a new more clean branch.
Thanks

### CODE REVIEW

File: src/common/config/public/ClientConfig.ts [hunk 1]
```diff
@@ -304,6 +304,16 @@ export class ClientSharingConfig {
...
   })
   passwordRequired: boolean = false;
+  @ConfigProperty({
+    type: 'unsignedInt', min: 8, max: 64,
+    tags:
+      {
+        name: $localize`Sharing key length`,
+        priority: ConfigPriority.underTheHood
+      },
+    description: $localize`The longer the keys are, the more secure the share links will be.`,
```
Comment (bpatrik) [inline src/common/config/public/ClientConfig.ts:314]: I would rather say something like: that longer the key, harder is it to guess it. The current phrasing implies better underlying security with longer keys.

I would also add that using lenght of 8 takes 70000 years to try all permutation (assuming the server can serve a request in 10s), which should be complex enough.

Also i would add a note that changing this number won't invalidate existing sharing

**Review by bpatrik — CHANGES_REQUESTED** [review]

**Comment by bpatrik** [comment]
Nice! Thank you! I made a comment on the config description. Otherwise it looks good to me.

File: src/common/config/public/ClientConfig.ts (hunk 1 above)
Comment (VirtualGit) [reply src/common/config/public/ClientConfig.ts:314]: Ok, I changed to "The longer the keys are, the harder they are to guess.".
Do you want I add "Changing this number won't invalidate existing sharing" in the setting description ?

Comment (VirtualGit) [reply src/common/config/public/ClientConfig.ts:314]: Also, should I do a new PR with these changes, or can I update this one ?

Comment (bpatrik) [reply src/common/config/public/ClientConfig.ts:314]: Yes please add that text too. 
Yes you should be able merge the changes here too.

**Comment by VirtualGit** [comment]
Sorry for the close, I had some troubles with PR. I'm just submitting a new one.

**Pull request closed without merging**